src/merge_data.rs

```diff
@@ -0,0 +1,454 @@
+            entry_info: EntryInfo::Panel {
+                short_name: "F".to_string(),
+                long_name: "First".to_string(),
+                summary: None,
+                slots: vec![
+                    EntryInfo::Slot {
+                        short_name: "S1".to_string(),
+                        long_name: "Slot 1".to_string(),
+                        max_rows: 1,
+                    },
+                    EntryInfo::Slot {
+                        short_name: "S2".to_string(),
+                        long_name: "Slot 2".to_string(),
+                        max_rows: 1,
+                    },
+                ],
+            },
+            interval: Interval::new(Timestamp(0), Timestamp(1000)),
+            tile_set: TileSet { tiles: Vec::new() },
+            field_schema: FieldSchema::new(),
+        };
+        let second = DataSourceInfo {
+            entry_info: EntryInfo::Panel {
+                short_name: "S".to_string(),
+                long_name: "Second".to_string(),
+                summary: None,
+                slots: vec![EntryInfo::Slot {
+                    short_name: "S3".to_string(),
+                    long_name: "Slot 3".to_string(),
+                    max_rows: 2,
+                }],
+            },
+            interval: Interval::new(Timestamp(0), Timestamp(2000)),
+            tile_set: TileSet { tiles: Vec::new() },
+            field_schema: FieldSchema::new(),
+        };
+
+        let infos = vec![first, second];
+
+        let mapping = MergeDeferredDataSource::compute_mapping(&infos);
+        assert_eq!(mapping, vec![0, 2]);
+
+        let merge = MergeDeferredDataSource::merge_infos(infos);
+
+        assert_eq!(merge.interval, Interval::new(Timestamp(0), Timestamp(2000)));
+        assert!(merge.tile_set.tiles.is_empty());
+
+        let EntryInfo::Panel {
+            short_name,
+            long_name,
+            summary,
+            slots,
+        } = merge.entry_info
+        else {
+            panic!("unexpected variant result in merge");
+        };
+
+        assert_eq!(short_name, "F");
+        assert_eq!(long_name, "First");
+        assert!(summary.is_none());
+        assert_eq!(slots.len(), 3);
+
+        let EntryInfo::Slot {
+            short_name: slot0_short_name,
+            ..
+        } = &slots[0]
+        else {
+            panic!("unexpected variant result in slot 0");
+        };
+
+        let EntryInfo::Slot {
+            short_name: slot1_short_name,
+            ..
+        } = &slots[1]
+        else {
+            panic!("unexpected variant result in slot 1");
+        };
+
+        let EntryInfo::Slot {
+            short_name: slot2_short_name,
+            ..
+        } = &slots[2]
+        else {
+            panic!("unexpected variant result in slot 1");
+        };
+
+        assert_eq!(slot0_short_name, "S1");
+        assert_eq!(slot1_short_name, "S2");
+        assert_eq!(slot2_short_name, "S3");
```

Comment:
My only suggestion would be to keep related cases entirely together rather than group by "operation"

```suggestion
        let EntryInfo::Slot {
            short_name: slot0_short_name,
            ..
        } = &slots[0]
        else {
            panic!("unexpected variant result in slot 0");
        };
        assert_eq!(slot0_short_name, "S1");

        let EntryInfo::Slot {
            short_name: slot1_short_name,
            ..
        } = &slots[1]
        else {
            panic!("unexpected variant result in slot 1");
        };
        assert_eq!(slot1_short_name, "S2");

        let EntryInfo::Slot {
            short_name: slot2_short_name,
            ..
        } = &slots[2]
        else {
            panic!("unexpected variant result in slot 1");
        };
        assert_eq!(slot2_short_name, "S3");
```

Reply:
oh and that last case should read "unexpected variant result in slot 2"